daemon/emer-main.c

```diff
@@ -186,8 +239,8 @@ on_authorize_method_check (GDBusInterfaceSkeleton *interface,
     g_dbus_method_invocation_return_error (invocation,
                                            G_DBUS_ERROR,
                                            G_DBUS_ERROR_AUTH_FAILED,
-                                           "Disabling metrics is only "
-                                           "allowed from system settings");
+                                           "%s",
+                                           authorized_method->error_message);
```

Comment:
If you were feeling really keen you could use `g_dbus_method_invocation_return_error_literal` and avoid the `"%s"`.